samples/siwx91x_ota/src/main.c

```diff
@@ -0,0 +1,1029 @@
+	       wifi_band_txt(entry->band), entry->rssi, wifi_security_txt(entry->security),
+	       ((entry->mac_length) ? mac_to_string(entry->mac, WIFI_MAC_ADDR_LEN, mac_string_buf,
+						    sizeof(mac_string_buf))
+				    : ""),
+	       wifi_mfp_txt(entry->mfp));
+}
+
+static void handle_wifi_scan_done(struct net_mgmt_event_callback *cb)
+{
+	const struct wifi_status *status = (const struct wifi_status *)cb->info;
+	struct app_ctx *app_ctx_st = CONTAINER_OF(cb, struct app_ctx, wlan_mgmt_cb);
+
+	if (status->status) {
+		printf("Scan request failed (%d)\n", status->status);
+	} else {
+		printf("Scan request done\n");
+		k_sem_give(&app_ctx_st->wlan_sem);
+	}
+	app_ctx_st->state = OTA_STATE_CONNECT;
+}
+
+static void handle_wifi_connect_result(struct net_mgmt_event_callback *cb)
+{
+	const struct wifi_status *status = (const struct wifi_status *)cb->info;
+	struct app_ctx *app_ctx_st = CONTAINER_OF(cb, struct app_ctx, wlan_mgmt_cb);
+
+	if (status->status) {
+		printf("Connection request failed (%d)\n", status->status);
+	} else {
+		printf("Connected to Wi-Fi\n");
+		k_sem_give(&app_ctx_st->wlan_sem);
+		app_ctx_st->state = OTA_STATE_IP_CONFIG;
+	}
+}
+
+static void handle_wifi_disconnect_result(struct net_mgmt_event_callback *cb)
+{
+	const struct wifi_status *status = (const struct wifi_status *)cb->info;
+	struct app_ctx *app_ctx_st = CONTAINER_OF(cb, struct app_ctx, wlan_mgmt_cb);
+
+	if (status->status) {
+		printf("Disconnection request failed (%d)\n", status->status);
+	} else {
+		printf("Disconnection reason: %d\n", status->disconn_reason);
+
+		/* Check if disconnection occurred during OTA process */
+		if (app_ctx_st->state > OTA_STATE_IP_CONFIG) {
+			printf("WiFi disconnected during OTA update, initiating reconnection\n");
+			ota_cleanup_resources(app_ctx_st);
+
+			/* Set state to reconnect */
+			app_ctx_st->state = OTA_STATE_CONNECT;
+		}
+		k_sem_give(&app_ctx_st->wlan_sem);
+	}
+}
+
+static void wifi_mgmt_event_handler(struct net_mgmt_event_callback *cb, uint64_t mgmt_event,
+				    struct net_if *iface)
+{
+
+	switch (mgmt_event) {
+	case NET_EVENT_WIFI_SCAN_RESULT:
+		handle_wifi_scan_result(cb);
+		break;
+	case NET_EVENT_WIFI_SCAN_DONE:
+		handle_wifi_scan_done(cb);
+		break;
+	case NET_EVENT_WIFI_CONNECT_RESULT:
+		handle_wifi_connect_result(cb);
+		break;
+	case NET_EVENT_WIFI_DISCONNECT_RESULT:
+		handle_wifi_disconnect_result(cb);
+		break;
+	default:
+		break;
+	}
+}
+
+int main(void)
+{
+	static struct app_ctx app_ctx = {
+		.wlan_sem = Z_SEM_INITIALIZER(app_ctx.wlan_sem, 0, 1),
+		.ota_sem = Z_SEM_INITIALIZER(app_ctx.ota_sem, 0, 1),
+		.dhcp_sem = Z_SEM_INITIALIZER(app_ctx.dhcp_sem, 0, 1),
+		.http_data_q = Z_MSGQ_INITIALIZER(app_ctx.http_data_q, app_ctx.http_data_q_buffer,
+						  sizeof(msg_t), HTTP_MSG_Q_MAX_SIZE),
+		.sock = -1,
+		.ota_range_end_byte = (CHUNK_SIZE - 1),
+		.state = OTA_STATE_SCAN};
+
+	app_ctx.iface = net_if_get_first_wifi();
```

Comment:
You don't need to store the `iface`. You can retrieve it with `net_if_get_first_wifi()` (single source of trust).